### DISCUSSION TITLE
Possible to include the source URL in email?

### DISCUSSION THREAD

**Discussion by nyonson** [slurdge/goeland#104] [Ideas]
Loving goeland, coming from rss2email. One feature I miss is having a link to the source article in the email. Is this possible with goeland or is it kinda fighting the sources/pipes model?

**Comment by nyonson** [comment]
Somewhat related (migrating from rss2email), was kinda hoping I could merge all these sources into one "feed" source and just have one "feed" pipe. But it looks like `SourceTitle` isn't available post-merge of sources, so for now I have a 1:1 relationship between source and pipe (little painful to add new sources).

```
[sources.emersion]
url = "https://emersion.fr/blog/atom.xml"
type = "feed"
filters = ["unseen", "lasthours(48)"]

[sources.drewdevault]
url = "https://drewdevault.com/blog/index.xml"
type = "feed"
filters = ["unseen", "lasthours(48)"]

[sources.rodarmor]
url = "https://rodarmor.com/blog/feed.xml"
type = "feed"
filters = ["unseen", "lasthours(48)"]

...

# SourceTitle is RSS feed title 
[pipes]

[pipes.emersion]
source = "emersion"
destination = "email"
email_to = ["nick@yonson.dev"]
email_from = "gemini2 <gemini@yonson.dev>"
email_title = "[RSS] {{.SourceTitle}} // {{.EntryTitle}}"

[pipes.drewdevault]
source = "drewdevault"
destination = "email"
email_to = ["nick@yonson.dev"]
email_from = "gemini2 <gemini@yonson.dev>"
email_title = "[RSS] {{.SourceTitle}} // {{.EntryTitle}}"

[pipes.rodarmor]
source = "rodarmor"
destination = "email"
email_to = ["nick@yonson.dev"]
email_from = "gemini2 <gemini@yonson.dev>"
email_title = "[RSS] {{.SourceTitle}} // {{.EntryTitle}}"

...
```

**Comment by slurdge** [comment]
> a link to the source article in the email.

I'm not sure I follow, would have an example of how it works ? It's like a source: [URL] added at the end of articles ? You already can include the link of articles and it will transform the title into a link back to the source entry.
So is it a link to the whole source (the rss feed) or each entry ?

For your second question, you want to have 3 sources and only one pipe ? If that's the case, what would be the title of such pipe ? The title of the first (merged) source ?

**Reply by nyonson** [reply]
For the first question, I was thinking a link to the entry. So in my scenario, it would be a link to the blog post (e.g. https://emersion.fr/blog/2023/status-update-49/). I think this would be easy enough to add to an email if the [`URL` of the entry](https://github.com/slurdge/goeland/blob/master/internal/goeland/source.go#L12) was exposed in the email template?

For the second question, I think my pipe could be called something generic like "Blog Feed". Ideally each new post entry would be sent with the blog source somehow. I think to support this, the `Entry` struct would need a new field to hold its original source post-aggregation(s).

**Reply by slurdge** [reply]
Can you try the `includelink` filter ? I still don't see what would be the end result, sorry.

**Reply by nyonson** [reply]
Ah, I didn't see the `includelink` option. The output with that is pretty much what I want, although you have to use it with the `digest` filter it seems. I'd rather have an email-per-entry, but I think I can just roll with the digest-per-source for now.

**Comment by slurdge** [comment]
This has been merged and will be available in next version.
